src/cmd/ping.rs

```diff
@@ -0,0 +1,70 @@
+use bytes::Bytes;
+use tracing::{debug, instrument};
+use crate::{Connection, Frame, Parse, ParseError};
+
+/// Returns PONG if no argument is provided, otherwise
+/// return a copy of the argument as a bulk.
+///
+/// This command is often used to test if a connection
+/// is still alive, or to measure latency.
+#[derive(Debug, Default)]
+pub struct Ping {
+    /// optional message to be returned
+    msg: Option<String>,
+}
+
+impl Ping {
+    /// Create a new `Ping` command with optional `msg`.
+    pub fn new(msg: Option<impl ToString>) -> Ping {
+        Ping {
+            msg: msg.map(|m| m.to_string()),
+        }
```

Comment:
There's no reason to do this.
```suggestion
    pub fn new(msg: Option<String>) -> Ping {
        Ping {
            msg,
        }
```